src/ui/SmartSnippet/SmartSnippet.ts

```diff
@@ -88,31 +122,31 @@ export class SmartSnippet extends Component {
   /**
    * @warning This method only works for the demo. In practice, the source of the answer will not always be part of the results.
    */
-  private getCorrespondingResult(questionAnswer: IQueryAnswerResponse) {
+  private getCorrespondingResult(questionAnswer: IQuestionAnswerResponse) {
     return find(
-      this.queryController.getLastResults().results,
+      this.lastResults.results,
       result => result.raw[questionAnswer.documentId.contentIdKey] === questionAnswer.documentId.contentIdValue
     );
   }
 
   private handleQuerySuccess(data: IQuerySuccessEventArgs) {
-    const { questionAnswer } = data.results;
+    const { questionAnswer } = (this.lastResults = data.results);
     if (!questionAnswer) {
       this.hasAnswer = false;
       return;
     }
     this.hasAnswer = true;
+    this.lastRenderedResult = this.getCorrespondingResult(questionAnswer);
```

Comment:
I would not make that a class property/adding that as class state to maintain.

I'd just create an utility function that scan `this.queryController.getLastResults()` and find the correct one (in a stateless manner)

Reply:
Good idea.

Looking back, it's no longer necessary to fetch the corresponding result in this function, that could be done inside the `render` function directly.